test/js/main.js

```diff
@@ -7,6 +7,13 @@ import {eachP,asyncForEach} from './utils/async'
 import {spawnP, spawnPrinter, SHELL_COLOR_BLUE} from './utils/misc'
 import {exec} from 'child_process'
 
+if (!process.env.BROWSERSTACK_USERNAME) {
```

Comment:
What is this? Did you branch off development? For some reason the PR was pointing at `master`

Reply:
Branched off `development`. I must've not updated the branch it was comparing to when i created the PR.

Anyway, as to that `BROWSERSTACK_USERNAME`, I probably shouldn't have made it part of this PR's commits. Just a check I added when I was trying to get UI tests running locally and didn't realise I needed to configure Browserstack credentials also.